feature/show api errors

ncocquio opened the pull request:
#9

Commit:
Add errors to all services

Commit:
Enable close button
- Add ids to toasts

kelvinlouis commented on src/actions/toast.actions.ts:
```diff
@@ -48,40 +49,47 @@ export interface ToastActions {
-    list: [...list.slice(0, -1)],
-  }),
+  hide: (id?: number) => ({ list }) => (
+    { list: [...list.filter(t => t.id !== id)] }
```
```
list: list.filter(t => t.id !== id)
```

kelvinlouis commented on src/services/EmployeeService.ts:
```diff
@@ -40,27 +80,154 @@ class EmployeeService {
-      .then((response: Contract) => new ContractModel(response));
+      .then((response: Contract) => new ContractModel(response))
+      .catch((error) => {
+        if (error.status === ResponseStatusCode.Unauthorized) {
```
Add the Unauthorized, InternalServerError and Network error checks in there.
Put it as a static function into ApiService.ts

```
checkDefaultResponseStatus(error);
```

Commit:
PR review fixes
Use bulma style close button
Toast messages without punct

Commit:
Move common errors to static function

Commit:
Fix lint errors

Commit:
Refactor form actions
- Move form actions in form-actions file
- Create RequestModel in Action, not Service

Commit:
Merge remote-tracking branch 'remotes/origin/develop' into feature/show-api-errors

Commit:
WIP: Moved functions in allocations
- Fixed typos and lint errors
- Add validation to allocation form
